preview-maker.js

```diff
@@ -196,6 +186,12 @@ const fill = (rootNode, color) => {
   innerFill(rootNode);
 };
 
+const isNotGrey = (hsl) =>
+  hsl.saturation > 0.05 && hsl.lightness < 0.92 && hsl.lightness > 0.08;
+
+const eqv = (x, y) =>
+  x.hue == y.hue && x.saturation == y.saturation && x.lightness == y.lightness;
```

Comment:
Floats are imprecise. For example, if you can compute the same value in two different ways, floats can give a small error, but that error is enough to make one value non-equal to the other. To avoid this, floats must be compared using epsilon comparison.